src/components/routing/routeProtectors/UserGuard.js

```diff
@@ -1,10 +1,26 @@
 import {Redirect} from "react-router-dom";
 import PropTypes from "prop-types";
 import Sidebar from "components/views/Sidebar";
+import {api} from 'helpers/api';
 
 export const UserGuard = props => {
-    let token = localStorage.getItem("token");
-    if (token && token !== 'undefined' && token != null && token !== '') {
+
+
+    // check if the token and the id match a user in the data base that is currently logged in
+    async function isTokenUserOnline() {
+        let token = localStorage.getItem("token");
+        let id = localStorage.getItem("id");
+        if (id && token && token !== 'undefined' && token != null && token !== '') {
+            // call api to check if the token and id match a user in the database and logged in
+            const response = await api.get(`/users/${id}`)
+            if (response.data.token === token && response.data.status === "ONLINE") {
```

Comment:
We never want to pass the token in the body (I'm not sure why, though), so you should probably pass the token via header (Do you also have a branch in the server?) I would not use get(`/users/${id}` as this is used by other users to get details about you (see userRepository.checkGeneralAccess()), but rather an extra API endpoint (that might return a boolean (server checks everything and just says yes or no)). [**MUST**]

Reply:
if you want to delete userguard, consider those comments here for the file below.